src/commands/new.rs

```diff
@@ -28,21 +44,39 @@ impl NewCommand {
         self.run_template(repo, id).await
     }
 
+    async fn list_templates(&self) -> anyhow::Result<()> {
+        let entries = hub_api::index().await.unwrap();
+        let matches = entries.iter()
+            .filter(|e| self.is_terms_match(e))
+            .sorted_by_key(|e| e.title())
+            .collect_vec();
+
+        if matches.is_empty() {
+            println!("No templates match your search terms");
+            return Ok(());
+        }
+
+        for entry in matches {
+            println!("Template: {}\nDescription: {}\n", entry.title(), entry.summary());
+        }
+
+        Ok(())
+    }
+
     async fn run_template(&self, repo: String, id: String) -> anyhow::Result<()> {
         use spin_templates::*;
 
-        let tempdir = tempfile::tempdir().unwrap();
-        let template_manager = spin_templates::TemplateManager::in_dir(tempdir.path());
+        let manager = TemplateManager::try_default()?;
 
         let source = TemplateSource::try_from_git(&repo, &None, &crate::spin::version())?;
         let options = InstallOptions::default();
-        template_manager.install(&source, &options, &DiscardingProgressReporter).await?;
+        manager.install(&source, &options, &DiscardingProgressReporter).await?;
 
-        let template = template_manager.get(&id).unwrap().unwrap();
+        let template = manager.get(&id).unwrap().unwrap();
```

Comment:
Rather than unwrapping, it would be better to propagate errors (I know I unwrapped, but that was wrong of me), so that Rust prints the error for the user rather than a panic backtrace.